modules/settings.js

```diff
@@ -13,63 +12,158 @@ function Settings() {
   };
 
   // Load options if present
-  (function() {
-    var retrievedOptions = localStorage.getItem('ffosbrOptions');
+  var retrievedOptions = localStorage.getItem('ffosbrOptions');
 
-    if (retrievedOptions !== null) {
-      self.options = JSON.parse(retrievedOptions);
-    }
-  })();
+  if (retrievedOptions !== null) {
 
-  Settings.prototype.options = function(myDictionary) {
-    if (!myDictionary) {
-      return options;
-    }
+    retrievedOptions = JSON.parse(retrievedOptions);
 
-    if (myDictionary.photos !== undefined &&
-      typeof myDictionary.photos === 'boolean') {
-      options.photos = myDictionary.photos;
+    if (this.validate(retrievedOptions === true)) {
+      this.options = retrievedOptions;
+    } else {
+      // TODO - should we throw an error? Or just let this slide?
+      console.log('Fetched an invalid options object from local storage');
     }
+  }
+}
 
-    if (myDictionary.videos !== undefined &&
-      typeof myDictionary.videos === 'boolean') {
-      options.videos = myDictionary.videos;
-    }
+Settings.prototype.validate = function(potentialOptions, value) {
 
-    if (myDictionary.contacts !== undefined &&
-      typeof myDictionary.contacts === 'boolean') {
-      options.contacts = myDictionary.contacts;
-    }
+  var valid = true;
+  var opts = null;
+  var validTypes = {
```

Comment:
Should this be declared outside this function so it doesn't get re-initialized with every call to validate()?